internal/build.go

```diff
@@ -95,6 +95,22 @@ func Build(repo *Repository, opts BuildOptions) error {
 
 		// See also `script` in Run.
 		shim := fmt.Sprintf(`#!/usr/bin/env node
+
+const { inspect } = require('util');
+process.on('uncaughtException', (exception) => {
+  process.stderr.write('uncaught exception: ' + inspect(exception) + '\n', () => {
+    process.exit(1);
+  });
+});
+process.on('unhandledRejection', (reason, promise) => {
+  process.stderr.write(
```

Comment:
Can't you just rethrow the error? Nbd.

Reply:
You mean so that `uncaughtException` can then run? I'm not precisely sure if that loses debugging context or not from having printed `reason` independently of `promise`.

Reply:
Yeah, it would allow error handling stuff to operate normally (e.g. Sentry)

Reply:
You always need some way to alert if there is a crash and the Sentry logic doesn't get to run. I should probably write this down somewhere, but an unhandled rejection should be considered a crash. Always use await and try/catch.

Reply:
> You always need some way to alert if there is a crash and the Sentry logic doesn't get to run
Sure — this just means it doesn't get to try. I'd argue that an unhandled rejection should be philosophically the same as an unhandled error (which does usually cause a crash but with a hook to allow recovery or logging).

Reply:
you don't get to try ... via the `uncaughtException` mechanism. instead, you have to do try/await/catch. I think that's a big improvement in terms of understandability.